homeassistant/components/faa_delays/binary_sensor.py

```diff
@@ -0,0 +1,94 @@
+"""Platform for FAA Delays sensor component."""
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.const import ATTR_ICON, ATTR_NAME
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, FAA_BINARY_SENSORS
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up a FAA sensor based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    binary_sensors = []
+    for kind, attrs in FAA_BINARY_SENSORS.items():
+        name = attrs[ATTR_NAME]
+        icon = attrs[ATTR_ICON]
+
+        binary_sensors.append(
+            FAABinarySensor(coordinator, kind, name, icon, entry.entry_id)
+        )
+
+    async_add_entities(binary_sensors)
+
+
+class FAABinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """Define a binary sensor for FAA Delays."""
+
+    def __init__(self, coordinator, sensor_type, name, icon, entry_id):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+
+        self.coordinator = coordinator
+        self._entry_id = entry_id
+        self._icon = icon
+        self._name = name
+        self._sensor_type = sensor_type
+        self._state = None
+        self._id = self.coordinator.data.iata
+        self._attrs = {}
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._id} {self._name}"
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
+
+    @property
+    def is_on(self):
+        """Return the status of the sensor."""
+        if self._sensor_type == "GROUND_DELAY":
+            self._state = self.coordinator.data.ground_delay.status
+        elif self._sensor_type == "GROUND_STOP":
+            self._state = self.coordinator.data.ground_stop.status
+        elif self._sensor_type == "DEPART_DELAY":
+            self._state = self.coordinator.data.depart_delay.status
+        elif self._sensor_type == "ARRIVE_DELAY":
+            self._state = self.coordinator.data.arrive_delay.status
+        elif self._sensor_type == "CLOSURE":
+            self._state = self.coordinator.data.closure.status
+        return self._state
```

Comment:
Looks like this doesn't need to be an instance attribute. It can just be a local variable.